server/src/main/java/io/druid/server/coordination/DruidServerMetadata.java

```diff
@@ -40,7 +43,10 @@ public DruidServerMetadata(
       @JsonProperty("maxSize") long maxSize,
       @JsonProperty("type") String type,
       @JsonProperty("tier") String tier,
-      @JsonProperty("priority") int priority
+      @JsonProperty("priority") int priority,
```

Comment:
"Having an interface which contains server metadata" seems reasonable.

"Having a java object which contains all types of server metadata regardless of how much of it is used for a service" does not seem reasonable.